iceoryx_hoofs/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -81,6 +76,14 @@ SharedMemory::~SharedMemory() noexcept
     destroy();
 }
 
+int SharedMemory::getOflagsFor(const AccessMode accessMode, const Policy policy) noexcept
+{
+    int oflags = 0;
+    oflags |= (accessMode == AccessMode::READ_ONLY) ? O_RDONLY : O_RDWR;
+    oflags |= (policy != Policy::OPEN) ? O_CREAT | O_EXCL : 0;
```

Comment:
Is this correct? Shouldn't `O_EXCL` only be set if the policy is `EXCLUSIVE_CREATE`?

Reply:
Actually no.
1. `PURGE_AND_CREATE` deletes the shared memory beforehand with `shm_unlink` and therefore we can assume that it does not exist. If its still existing either `shm_unlink` failed for unknown reasons or someone else swooped in and created a shared memory partition with that name before we could.
2. `OPEN_OR_CREATE` here we take ownership and my algorithm was like: create exclusively (and we have the ownership and have to clean up on destruction) or create fails since it already exists and we open it and do not have the ownership.

For `OPEN_OR_CREATE` I could also reverse the approach and open first and when it fails create the shared memory but the code looked a bit more cumbersome and therefore I decided for this approach.